crates/ruma-api-macros/src/request/outgoing.rs

```diff
@@ -15,39 +15,41 @@ impl Request {
 
         let method = &self.method;
         let error_ty = &self.error_ty;
-        let request_path_string = if self.has_path_fields() {
-            let mut format_string = self.path.value();
-            let mut format_args = Vec::new();
 
-            while let Some(start_of_segment) = format_string.find(':') {
-                // ':' should only ever appear at the start of a segment
-                assert_eq!(&format_string[start_of_segment - 1..start_of_segment], "/");
+        let (unstable_path, r0_path, stable_path) = if self.has_path_fields() {
+            let mapper = |s: &LitStr| -> TokenStream {
```

Comment:
I think this should just be called the same thing as the util function, `mapper` is a weird name.